### PR TITLE
isDeletePending should check the parents delete too

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Following up on PR: https://github.com/microsoft/FluidFramework/pull/14226
Seems like there was a bug where sudir.isSubDirectoryDeletePending should check in parent chain whether delete is pending or not and not just the immediate parent.

### CODE REVIEW

File: packages/dds/map/src/test/mocha/directory.spec.ts [hunk 1]
```diff
@@ -1108,7 +1184,7 @@ describe("Directory", () => {
 				);
 			});
 
-			it.skip("Directories should ensure eventual consistency using LWW approach 1", async () => {
+			it("Directories should ensure eventual consistency using LWW approach 1", async () => {
```
Comment (scarlettjlee) [inline packages/dds/map/src/test/mocha/directory.spec.ts:1187]: Remove or update the comment in this test (and the subsequent test) now that you've fixed the buggy behaviour described in the comment.

File: packages/dds/map/src/test/mocha/directory.spec.ts [hunk 2]
```diff
@@ -1142,7 +1218,7 @@ describe("Directory", () => {
...
```
Comment (scarlettjlee) [inline packages/dds/map/src/test/mocha/directory.spec.ts:1221]: Update the name/description of the test so it's not the same as the previous one and we can tell the results apart.

File: packages/dds/map/src/directory.ts [hunk 3]
```diff
@@ -743,29 +743,31 @@ export class SharedDirectory
...
-		return this.getWorkingDirectory(parentAbsPath) as SubDirectory;
+		let currentParent = this.root;
+		const nodeList = absolutePath.split(posix.sep);
+		let start = 1;
+		while (start < nodeList.length) {
+			const subDirName = nodeList[start];
+			if (currentParent.isSubDirectoryDeletePending(subDirName)) {
+				return true;
+			}
+			const newParent = currentParent.getSubDirectory(subDirName) as SubDirectory;
```
Comment (scarlettjlee) [inline packages/dds/map/src/directory.ts:763]: You could just assign to currentParent here.

File: packages/dds/map/src/directory.ts [hunk 4]
```diff
@@ -780,8 +782,8 @@ export class SharedDirectory
 				localOpMetadata,
 			) => {
 				const subdir = this.getWorkingDirectory(op.path) as SubDirectory | undefined;
-				// If there is pending delete op for this subDirectory, then don't apply the this op as we are going
-				// to delete this subDirectory.
+				// If there is pending delete op for any parent of this subdirectory, then don't apply the this op
```
Comment (scarlettjlee) [inline packages/dds/map/src/directory.ts:785]:
```suggestion
				// If there is pending delete op for this subdirectory or any parent, then don't apply the this op
```

It's ambiguous from the comment that the subdirectory itself is included. Same for other comments below.

**Review by scarlettjlee — APPROVED** [review]

File: packages/dds/map/src/directory.ts (hunk 3 above)
Comment (jatgarg) [reply packages/dds/map/src/directory.ts:763]: done

**Pull request merged**